server/chat-plugins/blackjack.js

```diff
@@ -0,0 +1,643 @@
+/**
+ * Blackjack Game
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ *
+ * This allows users to play the classic blackjack card game.
+ * Credits: jd, panpawn
+ *
+ * @license MIT license
+ */
+'use strict';
+
+class Blackjack extends Rooms.RoomGame {
+	constructor(room, user, target) {
+		super(room);
+		this.room = room;
+
+		this.turnTimeoutMinutes = 1;
+		this.timerTickSeconds = 5;
+
+		this.blackjackNumber = 0;
+		this.createdBy = user.name;
+		this.startedBy = '';
+		this.allowRenames = true;
+
+		this.playerCap = 16;
+		this.minimumPlayers = 2;
+		this.playerScrollWheel = 4;
+		this.cardWidth = 50;
+		this.cardHeight = 85;
+
+		this.spectators = Object.create(null);
+		this.dealer = new BlackjackDealer();
+
+		this.symbols = {
+			'♥': 'Hearts',
+			'♦': 'Diamonds',
+			'♣': 'Clubs',
+			'♠': 'Spades',
+		};
+		this.deck = new BlackjackDeck().shuffle();
+
+		this.id = this.room.id;
+		this.title = `Blackjack (${room.title})`;
+		this.blackjack = true;
+		this.state = 'signups';
+
+		this.lastMessage = '';
+		this.turnLog = '';
+		this.uhtmlChange = '';
+		this.curUser = '';
+		this.infoboxLimited = '';
+
+		this.button = '<button class="button" name="send" value="/joingame" title="Join Blackjack">Join</button> | <button class="button" name="send" value="/leavegame" title="Leave Blackjack">Leave</button>';
+		this.spectateButton = '<button class="button" name="send" value="/blackjack spectate" title="Spectate Blackjack">Spectate</button>';
+		this.slideButton = '<button class="button" name="send" value="/blackjack slide" title="Slide the game log down in the chat">(<i class="fa fa-arrow-down" aria-hidden="true"></i> slide)</button>';
+		this.atLeastOneJoin = false;
+
+		this.makeGame(target);
+	}
+
+	/**
+	 * Game Setup
+	 * makeGame - configures required settings for creating a game
+	 * makePlayer - adds blackjack-specific properties to player object
+	 * sendInvite - called when a game is created, or when a player joins/leaves
+	 */
+	makeGame(target) {
+		if (target) target = parseFloat(target);
+		if (isNaN(target) || target <= 0) target = '';
+		if (target !== '') {
+			this.autostart = setTimeout(() => this.start(), (target * 60000));
+		}
+		if (!this.room.blackjack) this.room.blackjack = 1;
+		this.blackjackNumber = this.room.blackjack;
+
+		this.sendInvite();
+		this.room.send(`|notify|Blackjack (${this.room.title})|${this.createdBy} created a game of Blackjack!`);
```

Comment:
im not sure if we want a room-wide declare every time a game starts

Reply:
Removed